tests/bindings/test_statevector_nb.py

```diff
@@ -0,0 +1,364 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for StateVector classes in nanobind-based modules."""
+
+import numpy as np
+import pytest
+from conftest import backend
+
+if backend != "qubit":
+    pytest.skip("Skipping tests for binaries other than lightning_qubit .", allow_module_level=True)
+
+
+class TestStateVectorNB:
+    """Tests for StateVectorC64 and StateVectorC128 classes in nanobind-based modules."""
+
+    # List of backends to test
+    backends = ["qubit", "kokkos", "gpu", "tensor"]
+
+    # List of modules to test
+    modules = [f"pennylane_lightning.lightning_{b}_nb" for b in backends]
+
+    @pytest.fixture
+    def get_statevector_class(self):
+        """Get StateVectorC64/128 class from module based on precision."""
+
+        def _get_class(module, precision="64"):
+            class_name = f"StateVectorC{precision}"
+            if hasattr(module, class_name):
+                return getattr(module, class_name)
+            pytest.skip(f"Class {class_name} not available in module")
+
+        return _get_class
+
+    @pytest.mark.parametrize("backend_name", backends)
+    def test_statevector_initialization(
+        self, backend_name, precision, nanobind_module, get_statevector_class
+    ):
+        """Test initialization of StateVectorC64/128 classes."""
+        module = nanobind_module(backend_name)
+
+        # Get the appropriate StateVector class based on precision
+        StateVectorClass = get_statevector_class(module, precision)
+
+        # Test initialization with number of qubits and updateData
+        num_qubits = 3
+        # Create a numpy array representing |0> state with appropriate size
+        dtype = np.complex128 if precision == "128" else np.complex64
+        state_data = np.zeros(2**num_qubits, dtype=dtype)
+        state_data[0] = 1.0
+
+        # Initialize with number of qubits first
+        sv = StateVectorClass(num_qubits)
+        # Then update data
+        sv.updateData(state_data)
+
+        # Check that the size is correct
+        assert sv.size() == 2**num_qubits
+
+    @pytest.mark.parametrize("backend_name", backends)
+    def test_statevector_methods(
```

Comment:
Is this meant to do something extra compared to `test_statevector_initialization`?

Reply:
Was, but with time, they evolved to be the same. I just removed.